.github/workflows/ci_cd.yml

```diff
@@ -0,0 +1,76 @@
+name: CI/CD
+
+on:
+  push:
+    branches: 
+    - master
+  pull_request:
+    branches: 
+    - master
+
+jobs:
+  test:
+    name: Test on node ${{ matrix.node_version }} and ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        node_version:
+        - 10
+        - 12
+        os: 
+        - ubuntu-latest
+        - windows-latest
+        - macOS-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Setting up Node.js ${{ matrix.node_version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node_version }}
+
+    - name: Get npm cache directory
+      id: npm-cache
```

Comment:
What does `id` do here? Just asking because I am unsure of its purpose.

Reply:
According to the documentation, the use of id is unnecessary for us [link for the purpose of id](https://github.com/actions/cache#skipping-steps-based-on-cache-hit). But if we don't use id then  CI will fail at step "Run actions/cache@v1" and also the documentation given uses id [link](https://github.com/actions/cache/blob/master/examples.md#node---npm)